src/slurm_plugin/clustermgtd.py

```diff
@@ -531,42 +516,30 @@ def _get_node_info_from_partition(partitions):
             log.error("Failed when getting partition/node states from scheduler with exception %s", e)
             raise ClusterManager.SchedulerUnavailable
 
-    def _clean_up_inactive_partition(self, inactive_nodes, cluster_instances):
+    def _clean_up_inactive_partition(self, inactive_nodes, inactive_instance_ids):
         """Terminate all other instances associated with nodes in INACTIVE partition directly through EC2."""
         try:
             log.info("Cleaning up INACTIVE partitions.")
-            private_ip_to_instance_map = {instance.private_ip: instance for instance in cluster_instances}
-            instances_to_terminate = ClusterManager._get_backing_instance_ids(
-                inactive_nodes, private_ip_to_instance_map
-            )
             log.info(
                 "Clean up instances associated with nodes in INACTIVE partitions: %s",
-                print_with_count(instances_to_terminate),
+                print_with_count(inactive_instance_ids),
             )
-            if instances_to_terminate:
+            if inactive_instance_ids:
                 self._instance_manager.delete_instances(
-                    instances_to_terminate, terminate_batch_size=self._config.terminate_max_batch_size
+                    inactive_instance_ids, terminate_batch_size=self._config.terminate_max_batch_size
                 )
 
             self._reset_nodes_in_inactive_partitions(inactive_nodes)
-
-            instances_still_in_cluster = []
-            for instance in cluster_instances:
-                if instance.id not in instances_to_terminate:
-                    instances_still_in_cluster.append(instance)
-
-            return instances_still_in_cluster
         except Exception as e:
             log.error("Failed to clean up INACTIVE nodes %s with exception %s", print_with_count(inactive_nodes), e)
-            return cluster_instances
 
     @staticmethod
     def _reset_nodes_in_inactive_partitions(inactive_nodes):
         # Try to reset nodeaddr if possible to avoid potential problems
         nodes_to_reset = []
         for node in inactive_nodes:
             if node.is_nodeaddr_set() or (
-                not node.is_static and not (node.is_power() or node.is_powering_down() or node.is_down())
+                not isinstance(node, StaticNode) and not (node.is_power() or node.is_powering_down() or node.is_down())
```

Comment:
Can we port this logic into the `Static/DynamicNode` class to avoid doing the `not isinstance(node, StaticNode)` check?
We can define a new function `needs_reset_when_inactive`(sorry for the bad name) and just call that here